src/Lecture2.hs

```diff
@@ -185,7 +298,15 @@ False
 True
 -}
 isIncreasing :: [Int] -> Bool
-isIncreasing = error "TODO"
+isIncreasing [] = True
+isIncreasing (x:xs) = isLess x xs
+  where
+  isLess :: Int -> [Int] -> Bool
+  isLess _ [] = True
+  isLess n (x: xs)
+    | n > x = False
+    | otherwise = isLess x xs
```

Comment:
You can implement `isIncreasing` in a slightly more elegant way by relying on laziness and nested pattern matching in Haskell 🦥 

```suggestion
isIncreasing [_] = True
isIncreasing (x : y : ys) = x < y && isIncreasing (y : ys)
```